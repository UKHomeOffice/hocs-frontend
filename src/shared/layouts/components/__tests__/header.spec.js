import React from "react";
import Header from "../header.jsx";

describe('Layout header component', () => {
    it('should render with default props', () => {
        expect(
            render(<Header/>)
        ).toMatchSnapshot();
    });
    it('should render the menu when passed', () => {
        const props = {
            menu: [
                {label: 'first', target: '/'},
                {label: 'second', target: '/'}
            ]
        };
        expect(
            shallow(<Header {...props}/>)
        ).toMatchSnapshot();
    });
});