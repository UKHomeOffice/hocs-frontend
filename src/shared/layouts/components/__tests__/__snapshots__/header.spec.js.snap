// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout header component should render the menu when passed 1`] = `
<React.Fragment>
  <header
    className="with-proposition"
    id="global-header"
    role="banner"
  >
    <div
      className="header-wrapper"
    >
      <div
        className="header-global"
      >
        <div
          className="header-logo"
        >
          <a
            className="content"
            href="https://www.gov.uk"
            id="header-application"
            title=""
          >
            GOV.UK
          </a>
        </div>
      </div>
      <div
        className="header-proposition"
      >
        <div
          className="content"
        >
          <nav
            id="proposition-menu"
          >
            <div
              id="proposition-header"
            >
              <a
                href="/"
                id="proposition-name"
              >
                Service Name
              </a>
            </div>
            <React.Fragment>
              <label
                aria-label="menu"
                htmlFor="toggle-mobile-menu"
              >
                Menu
              </label>
              <input
                id="toggle-mobile-menu"
                type="checkbox"
              />
              <ul
                id="main-menu"
              >
                <li
                  key="first"
                >
                  <Link
                    replace={false}
                    to="/"
                  >
                    first
                  </Link>
                </li>
                <li
                  key="second"
                >
                  <Link
                    replace={false}
                    to="/"
                  >
                    second
                  </Link>
                </li>
              </ul>
            </React.Fragment>
          </nav>
        </div>
      </div>
    </div>
  </header>
  <div
    id="global-header-bar"
  />
</React.Fragment>
`;

exports[`Layout header component should render with default props 1`] = `
Array [
  <header
    class="with-proposition"
    id="global-header"
    role="banner"
  >
    <div
      class="header-wrapper"
    >
      <div
        class="header-global"
      >
        <div
          class="header-logo"
        >
          <a
            class="content"
            href="https://www.gov.uk"
            id="header-application"
            title=""
          >
            GOV.UK
          </a>
        </div>
      </div>
      <div
        class="header-proposition"
      >
        <div
          class="content"
        >
          <nav
            id="proposition-menu"
          >
            <div
              id="proposition-header"
            >
              <a
                href="/"
                id="proposition-name"
              >
                React Application
              </a>
            </div>
          </nav>
        </div>
      </div>
    </div>
  </header>,
  <div
    id="global-header-bar"
  />,
]
`;
